src/Engine/CMakeLists.txt

```diff
@@ -35,6 +35,10 @@ if (ENABLE_CLANG_TIDY)
     set_target_properties(McBopomofoLMLib PROPERTIES CXX_CLANG_TIDY "${CLANG_TIDY_COMMAND}")
 endif ()
 
+if (ENABLE_EXPERIMENTAL_SIMD_SUPPORT_AVX512)
```

Comment:
maybe add a compiler support check to inform the user?

```cmake
include(CheckCXXCompilerFlag)

check_cxx_compiler_flag("-mavx512f" HAS_AVX512F)

if(HAS_AVX512F)
  message(STATUS "AVX-512 support detected")
  target_compile_options(McBopomofoLMLib PRIVATE -mavx512f -march=native -DENABLE_EXPERIMENTAL_SIMD_SUPPORT_AVX512=1)
else()
  message(STATUS "AVX-512 support not detected")
endif()
```

https://cmake.org/cmake/help/latest/module/CheckCXXCompilerFlag.html

Reply:
I'll make a note myself to investigate the compile-time check further. There's also this question about cross-building vs building-for-yourself, i.e. you may be able to build, but you can't run the built binary if the processor doesn't support it. Some libraries also use runtime dispatch to widen the support tiers. All for future works. :)